### PR TITLE
Add DisableIncrementalFolderCleanup option

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/LinkPhaseHandlerWithIncremental.cs [hunk 1]
```diff
@@ -321,19 +321,26 @@ private ManifestItem UpdateItem(ManifestItem item, string sourceRelativePath)
             var result = item.Clone();
             result.IsIncremental = true;
             result.SourceRelativePath = sourceRelativePath;
-            foreach (var ofi in result.OutputFiles.Values)
+
+            // Copy when current base dir is not last base dir
+            if (!FilePathComparer.OSPlatformSensitiveRelativePathComparer.Equals(
+                Environment.ExpandEnvironmentVariables(IncrementalContext.BaseDir),
+                Environment.ExpandEnvironmentVariables(IncrementalContext.LastBaseDir)))
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/LinkPhaseHandlerWithIncremental.cs]: how about extract a utility method?

**Comment by ansyral** [comment]
:shipit:

File: src/docfx/Models/BuildCommandOptions.cs [hunk 2]
```diff
@@ -117,5 +117,8 @@ internal class BuildCommandOptions : LogOptions, ICanPrintHelpMessage
 
         [Option("keepFileLink", HelpText = "If set to true, docfx does not dereference (aka. copy) file to the output folder, instead, it saves a link_to_path property inside mainfiest.json to indicate the physical location of that file.")]
         public bool KeepFileLink { get; set; }
+
+        [Option("disableIncrementalFolderCleanup", HelpText = "If set to true, docfx reuses last incremental cache folder instead of creating a new one. This can improve the performance of incremental build by not copying massive intermediate files, however, it introduces issue that historical files not cleaned up and thus the size of intermeidate folder is not under control.")]
+        public bool? DisableIncrementalFolderCleanup { get; set; }
```
Comment (hellosnow) [inline src/docfx/Models/BuildCommandOptions.cs]: >DisableIncrementalFolderCleanup [](start = 21, length = 31)

naming: how about 'KeepIntermediateFolder'?

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**